### PR TITLE
Create price list service

### PR DESCRIPTION
**Pull request by MrDecart**

### CODE REVIEW

File: VetClinicApi.Application/Services/PriceListHandling/PriceListService.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        public async Task DeletePriceList(int id)
+        {
+            try
+            {
+                await _priceListRepository.Delete(id);
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                throw new EntityNotFoundException(id, "priceList");
```
Comment (VintageTrouble) [inline VetClinicApi.Application/Services/PriceListHandling/PriceListService.cs]:
```suggestion
                throw new EntityNotFoundException(id, "PriceList");
```

File: VetClinicApi.Application/Services/PriceListHandling/PriceListService.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            if (priceList == null)
+                throw new ArgumentNullException(nameof(priceList), "PriceList can't be null.");
+
+            try
+            {
+                return await _priceListRepository.Update(priceList);
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                throw new EntityNotFoundException(priceList.Id, "priceList");
```
Comment (VintageTrouble) [inline VetClinicApi.Application/Services/PriceListHandling/PriceListService.cs]:
```suggestion
                throw new EntityNotFoundException(priceList.Id, "PriceList");
```

File: VetClinicApi.Application/Services/PriceListHandling/IPriceListService.cs
Comment (VintageTrouble) [inline VetClinicApi.Application/Services/PriceListHandling/IPriceListService.cs:1]: Suggestion: Probably we should use `PriceList` as many prices, but `Price` as one item?

File: VetClinicApi.Application.Tests/Services/PriceListHandling/PriceListService_Test.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+        await Assert.ThrowsAsync<EntityNotFoundException>(async () => await priceListService.DeletePriceList(1));
+    }
+
+    [Fact]
+    public async Task Update_PriceListIsNull_Test()
+    {
+        var priceListService = new PriceListService(_priceListRepository.Object);
+
+        await Assert.ThrowsAsync<ArgumentNullException>(async () => await priceListService.UpdatePriceList(null));
```
Comment (VintageTrouble) [inline VetClinicApi.Application.Tests/Services/PriceListHandling/PriceListService_Test.cs]: remove warning: "Cannot convert null literal to non-nullable reference type."

File: VetClinicApi.Application.Tests/Services/PriceListHandling/PriceListService_Test.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+public class PriceListService_Test
+{
+    private readonly Mock<IPriceListRepository> _priceListRepository = new();
+
+    [Fact]
+    public async Task Create_PriceListIsNull_Test()
+    {
+        var priceListService = new PriceListService(_priceListRepository.Object);
+
+        await Assert.ThrowsAsync<ArgumentNullException>(async () => await priceListService.CreatePriceList(null));
```
Comment (VintageTrouble) [inline VetClinicApi.Application.Tests/Services/PriceListHandling/PriceListService_Test.cs]: remove warning: "Cannot convert null literal to non-nullable reference type."

File: VetClinicApi.Application/Services/PriceListHandling/IPriceListService.cs
Comment (MrDecart) [reply VetClinicApi.Application/Services/PriceListHandling/IPriceListService.cs:1]: We decided to use "ProvidedService".

**Review by VintageTrouble — APPROVED** [review]

**Pull request merged**